common/console.cpp

```diff
@@ -404,7 +404,7 @@ namespace console {
                     } while (count == 0 && !widths.empty());
                 }
             } else {
-                int offset = line.length();
+                int offset = int(line.length());
```

Comment:
What is the point of this change (and many other similar changes)?

Are we seriously considering that one day we will be processing lines longer than `std::numeric_limits<int>::max()`? But suppose we are planning for a distant future where computers have become so powerful that it is in fact possible to process such humongous lines at all, and also do it in a reasonable amount of time. What would be the intent in such a hypothetical scenario? To silently truncate the line to  `std::numeric_limits<int>::max()`? If you look at the context rather than blindly applying suggestions of a static code analyzer, you will see that `offset` is being used to determine the amount of bytes added by the call to `append_utf8()`, so after you truncated to the max integer, we are having an actual bug for the case `line.length() > std::numeric_limits<int>::max()`. So, then, you either leave the way it is knowing that we will never ever be processing lines longer than `2^31` bytes, or you change it to `auto offset = line.length()` or `size_t offset = line_length()`.

Reply:
I strongly believe, that the code should compile without any, or with a limited amount of warnings. Otherwise the build log is heavily polluted with unrelated information and some genuine warnings may be lost.

As I mentioned previously, the most correct way would be to check and fix all the declarations and function parameters, but that would result in an even bigger PR and would take much more time.

Reply:
Well, but you did not achieve anything here. You removed a warning. If you want to change it, then it is `auto offset = line.length()`, not `int offset = int(line.length())`. 

Concerning warnings: on Linux the code is compiled with (almost) zero warning using `-Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -Wdouble-promotion`, which goes well beyond what I personally consider reasonable. If your compiler still produces tons of warnings, perhaps it is time to change the compiler or disable some of them? (which is the very first thing I have to deal with MSVC, for example)